packages/amplify-category-geo/src/service-walkthroughs/mapWalkthrough.ts

```diff
@@ -0,0 +1,199 @@
+        const mapNameInput = await prompter.input(
+            'Provide a name for the Map:',
+            { validate: alphanumeric(), initial: `map${shortId}` }
+        );
+        if (await checkGeoResourceExists(mapNameInput)) {
+            printer.info(`Map ${mapNameInput} already exists. Choose another name.`);
+        }
+        else mapName = mapNameInput;
+    }
+    return { name: mapName };
+};
+
+export const mapAdvancedWalkthrough = async (context: $TSContext, parameters: Partial<MapParameters>): Promise<Partial<MapParameters>> => {
+    // const includePricingPlan = await geoServiceExists(ServiceName.Map) || await geoServiceExists(ServiceName.PlaceIndex);
+    const includePricingPlan = false;
+    const currentPricingPlan = parameters.pricingPlan ? parameters.pricingPlan : await getGeoPricingPlan();
+    const advancedSettingOptions: string[] = ['Map style & Map data provider (default: Streets provided by Esri)'];
+    if (includePricingPlan) {
+        advancedSettingOptions.push(`Map pricing plan (current: ${currentPricingPlan})`);
+    }
+    printer.info('Available advanced settings:');
+    formatter.list(advancedSettingOptions);
+    printer.blankLine();
+
+    if(await prompter.yesOrNo('Do you want to configure advanced settings?', false)) {
+        // get the map style parameters
+        parameters = merge(parameters, await mapStyleWalkthrough(parameters));
+
+        if (includePricingPlan) {
+            // get the pricing plan
+            parameters = merge(parameters, await pricingPlanWalkthrough(context, parameters));
+        }
+        else {
+            parameters.pricingPlan = currentPricingPlan;
+        }
+    }
+    else {
+        parameters.dataProvider = DataProvider.Esri;
+        parameters.mapStyleType = EsriMapStyleType.Streets;
+        parameters.pricingPlan = currentPricingPlan;
+    }
+
+    return parameters;
+};
+
+export const mapStyleWalkthrough = async (parameters: Partial<MapParameters>): Promise<Partial<MapParameters>> => {
+    const mapStyleChoices = [
+        { name: 'Streets (data provided by Esri)', value: MapStyle.VectorEsriStreets },
+        { name: 'Berlin (data provided by HERE)', value: MapStyle.VectorHereBerlin },
+        { name: 'Topographic (data provided by Esri)', value: MapStyle.VectorEsriTopographic },
+        { name: 'Navigation (data provided by Esri)', value: MapStyle.VectorEsriNavigation },
+        { name: 'LightGrayCanvas (data provided by Esri)', value: MapStyle.VectorEsriLightGrayCanvas },
+        { name: 'DarkGrayCanvas (data provided by Esri)', value: MapStyle.VectorEsriDarkGrayCanvas }
+    ];
+    const mapStyleDefault = parameters.dataProvider && parameters.mapStyleType ?
+        getGeoMapStyle(parameters.dataProvider, parameters.mapStyleType) : 'VectorEsriStreets';
+
+    const mapStyleDefaultIndex = mapStyleChoices.findIndex(item => item.value === mapStyleDefault);
+    const mapStyleInput = await prompter.pick<'one', string>(
```

Comment:
the interface should let you type this as `<'one', MapStyle>` and then you don't have to do a type assertion on the return type

Reply:
I tried this. It throws an error like:
```
Argument of type '{ name: string; value: MapStyle; }[]' is not assignable to parameter of type 'string[] | GenericChoice<MapStyle.VectorEsriNavigation>[] | GenericChoice<MapStyle.VectorEsriStreets>[] | GenericChoice<MapStyle.VectorEsriTopographic>[] | GenericChoice<MapStyle.VectorEsriDarkGrayCanvas>[] | GenericChoice<...>[] | GenericChoice<...>[]'.
  Type '{ name: string; value: MapStyle; }[]' is not assignable to type 'string[]'.
    Type '{ name: string; value: MapStyle; }' is not assignable to type 'string'
```

Reply:
hmm okay there might be some problem with the function types then. I'll check on that, but this is fine for now